business/workloads.go

```diff
@@ -1062,6 +1082,53 @@ func fetchWorkload(layer *Layer, namespace string, workloadName string) (*models
 	return wl, kubernetes.NewNotFound(workloadName, "Kiali", "Workload")
 }
 
+func updateWorkload(layer *Layer, namespace string, workloadName string, jsonPatch string) error {
+	// Check if user has access to the namespace (RBAC) in cache scenarios and/or
+	// if namespace is accessible from Kiali (Deployment.AccessibleNamespaces)
+	if _, err := layer.Namespace.GetNamespace(namespace); err != nil {
+		return err
+	}
+
+	workloadTypes := []string{
+		kubernetes.DeploymentType,
+		kubernetes.ReplicaSetType,
+		kubernetes.ReplicationControllerType,
+		kubernetes.DeploymentConfigType,
+		kubernetes.StatefulSetType,
+		kubernetes.JobType,
+		kubernetes.CronJobType,
+		kubernetes.PodType,
+	}
```

Comment:
If I understand correctly, for a single workload patch request we're going to try to patch every possible kind of workloads with that name?
I guess it works in general because only one type will match with the provided name and the others will return errors / not found?

Is it possible to improve this, and provide the correct type directly from the query params / ui ?
This would give better performances (avoid unnecessary kube queries) and also prevent the corner case where several workloads of different types would have the same name (I guess this corner case would trigger patch errors)

Reply:
In Kiali we are assuming that we won't have two different controllers with same name in the namespace.
Basically, Istio didn't support that cornercase and we maintained that:
https://github.com/kiali/kiali/blob/master/business/workloads.go#L1065

Yes, perhaps in the handler we could pass an optional parameter with the workloadType and if present to follow that, otherwise fall into fetching all.

But this is a helper, to proper fix this we should revisit the workload model, and I tried to avoid that as it may trigger more major changes that perhaps are better to adress them in a next major version.

Reply:
I agree with @jotak. It looks like we can easily avoid the multiple calls to k8s API with passing this extra param. I see that the workload model on the backend already have the type field that can be useful in here.
I might be missing something over here though.

Reply:
Yes, an extra param may help here, but just keeping optional to not break the current model.
In short, kubernetes delegates into the user to not have two different controllers with same name, Istio followed that implicitly in their model and we mapped the workloads using that assumption.
But adding a check for get/update a single workload with a type may help here.
I'll add it during the week to the PR.

What I don't want to open is to introduce any kind of big refactoring as adding constratints into the workload model in Kiali for now, that's something we can discuss in a major iteration.